homeassistant/components/intellifire/coordinator.py

```diff
@@ -34,27 +30,37 @@ def __init__(
             name=DOMAIN,
             update_interval=timedelta(seconds=15),
         )
-        self._read_api = read_api
-        self._control_api = control_api
+        self._api = api
 
     async def _async_update_data(self) -> IntellifirePollData:
-        LOGGER.debug("Calling update loop on IntelliFire")
-        async with timeout(100):
-            try:
-                await self._read_api.poll()
-            except (ConnectionError, ClientConnectionError) as exception:
-                raise UpdateFailed from exception
-        return self._read_api.data
+
+        if not self._api.is_polling_in_background:
+            LOGGER.info("Starting Intellifire Background Polling Loop")
+            await self._api.start_background_polling()
+
+            # Don't return uninitialized poll data
+            async with timeout(100):
```

Comment:
100 is a bit much no?

Reply:
I'll lower it.

In some cases its actually not too low at all :(